src/nats.h

```diff
@@ -1243,6 +1278,44 @@ typedef struct jsOptions
 
         } PublishAsync;
 
+        struct jsOptionsPullSubscribeAsync
+        {
+                // Lifetime of the subscription (completes when any one of the
+                // targets is reached).
+                int64_t                 Timeout; // in milliseconds
+                int                     MaxMessages;
+                int64_t                 MaxBytes;
+
+                // If NoWait is set, the subscription will receive the messages
+                // already stored on the server subject to the limits, but will
+                // not wait for more messages.
+                bool                    NoWait;
```

Comment:
this should be not exposed as a public API.
Rather have two different methods for fetching with and without wait.

The reason being - expires and no wait has different behaviors:
expires set + no_wait - will try to get all messages up to the batch, or the number currently available
no expires + no_wait - will always wait for first message. Even if there are currently any on the server.

Reply:
As discussed offline, added an extra comment to clarify, but keeping the flag.